config/config.go

```diff
@@ -299,6 +308,7 @@ func (c *Config) UnmarshalJSON(data []byte) error {
 	c.EnableWebProfile = conf.EnableWebProfile
 	c.LogConfig = conf.LogConfig
 	c.Revision = conf.Revision
+	c.MaintenanceConfig = conf.MaintenanceConfig
```

Comment:
add a test for this in config_test.go/reader_test.go. should test marshalling and unmarshalling

Reply:
should validate that the resource name is actually a resource name

Reply:
I added a test for marshaling and unmarshaling the config. What do you mean by validate the resource name is actually a resource name?

Reply:
for some extra context I switched maintenance config's sensor name field to string because in the config we get from app sensor name is a string.

Reply:
> for some extra context I switched maintenance config's sensor name field to string because in the config we get from app sensor name is a string.

what do you mean by that - the config that gets sent over proto?

>  What do you mean by validate the resource name is actually a resource name?

There's an Ensure function that makes sure the config is valid. Can see a path where the name is not actually a name and so we just delete/remove the maintenance config from the config

Reply:
testing Marshal and Unmarshal by itself is good, and a good additional test would be to add to TestStoreToCache and test that the whole flow succeeds